lib/data-types.d.ts

```diff
@@ -396,105 +512,128 @@ export class VIRTUAL<T> extends ABSTRACT {
  *
  * @property ENUM
  */
-export class ENUM<T extends string> extends ABSTRACT {
-  constructor(...values: string[]);
-  constructor(options: ENUM_Options<T>);
-  values: T[];
-  options: ENUM_Options<T>;
+export const ENUM: EnumDataTypeStatic;
+
+export interface EnumDataTypeStatic extends AbstractDataTypeStatic {
+  new (...values: string[]): EnumDataType;
+  new (options: EnumDataTypeOptions): EnumDataType;
+  (...values: string[]): EnumDataType;
+  (options: EnumDataTypeOptions): EnumDataType;
 }
-export interface ENUM_Options<T extends string> {
-  values: T[];
+
+export interface EnumDataType extends AbstractDataType {
+  values: string[];
+  options: EnumDataTypeOptions;
+}
+
+export interface EnumDataTypeOptions {
+  values: string[];
 }
 
 /**
  * An array of `type`, e.g. `DataTypes.ARRAY(DataTypes.DECIMAL)`. Only available in postgres.
  * @property ARRAY
  */
-export class ARRAY<T extends new () => ABSTRACT> extends ABSTRACT {
-  constructor(type: T);
-  constructor(options: ARRAY_Options<T>);
+export const ARRAY: ArrayDataTypeStatic;
 
-  options: ARRAY_Options<T>;
+export interface ArrayDataTypeStatic extends AbstractDataTypeStatic {
+  new <T extends AbstractDataTypeStatic|AbstractDataType>(type: T): ArrayDataType<T>;
+  new <T extends AbstractDataTypeStatic|AbstractDataType>(options: ArrayDataTypeOptions<T>): ArrayDataType<T>;
+  <T extends AbstractDataTypeStatic|AbstractDataType>(type: T): ArrayDataType<T>;
+  <T extends AbstractDataTypeStatic|AbstractDataType>(options: ArrayDataTypeOptions<T>): ArrayDataType<T>;
+  is<T extends AbstractDataTypeStatic|AbstractDataType>(obj: any, type: T): obj is ArrayDataType<T>;
+}
 
-  static is<C>(obj: any, type: C): obj is ARRAY<C>;
+export interface ArrayDataType<T extends AbstractDataTypeStatic|AbstractDataType> extends AbstractDataType {
+  options: ArrayDataTypeOptions<T>;
 }
-export interface ARRAY_Options<T> {
+
+export interface ArrayDataTypeOptions<T extends AbstractDataTypeStatic|AbstractDataType> {
   type: T;
 }
 
 /**
  * A geometry datatype represents two dimensional spacial objects.
  * @property GEOMETRY
  */
-export class GEOMETRY extends ABSTRACT {
-  constructor(type, srid);
-  constructor(options: GEOMETRY_Options);
+export const GEOMETRY: GeometryDataTypeStatic;
 
-  options: GEOMETRY_Options;
-  type: any;
-  srid: any;
+export interface GeometryDataTypeStatic extends AbstractDataTypeStatic {
+  new (type: string, srid?: number): GeometryDataType;
+  new (options: GeometryDataTypeOptions): GeometryDataType;
+  (type: string, srid?: number): GeometryDataType;
+  (options: GeometryDataTypeOptions): GeometryDataType;
+}
 
+export interface GeometryDataType extends AbstractDataType {
+  options: GeometryDataTypeOptions;
+  type: string;
+  srid?: number;
   escape: boolean;
-  $stringify(value: any): string;
```

Comment:
my concern was that if someone is introducing a new type that extends one of the built-in types they should have access to those functions. I couldn't find anyone doing that anyway so it's fine.